lib/task-service.js

```diff
@@ -4,21 +4,54 @@
 
 var di = require('di');
 
-module.exports = factory;
-di.annotate(factory, new di.Provide('Services.Task'));
-di.annotate(factory, new di.Inject(
+module.exports = taskServiceFactory;
+di.annotate(taskServiceFactory, new di.Provide('Services.Task'));
+di.annotate(taskServiceFactory, new di.Inject(
+    'Logger',
+    'Promise',
+    'ChildProcess',
     'TaskOption.Validator'
 ));
 
-function factory(validator) {
+function taskServiceFactory(
+    Logger,
+    Promise,
+    ChildProcess,
+    validator
+) {
+    var logger = Logger.initialize(taskServiceFactory);
+    var requiredCmdTools = ['ipmitool'];
```

Comment:
As I said in the PR comment, I purposely only adding the ipmitool and I see the snmp and racadm  tool aren't used too often, especially the racadm tool is only used for Dell server and a few tasks, it's even not easy to install it. So if preventing user to use RackHD if not installing some not required tool (for user's special user case) doesn't make sense.

But I'm OK to include more if you think that's necessary.

As for why not define it as constant, since I thought the `requiredCmdTools` vary with different customers, so maybe it's better to be a configuration item or even get some input from skupack (like the racadm tool for Dell Servers).

Do you have any good idea to handle this?